Headless working with grasp_planning_graspit

JenniferBuehler opened the pull request:
This works with EGPlanner from [grasp_planning_graspit](https://github.com/JenniferBuehler/graspit-pkgs/tree/master/grasp_planning_graspit). Not many changes required actually. There are some minor changes in the CMakeLists.txt and I also added a graspitConfig.cmake so that I could compile grasp_planning_graspit - this should maybe go in a separate PR, but I added it here now in any case as it's quite straight forward. If you want, we can pluck it apart again.

Some of the #include tags still had to be fixed as well for compiling from my package.

Also: is it ok to be able to pass a world name into emptyWorld()?

Only issue left: The camera parameters. World files saved in headless mode cannot be opened with graspit_simulator, because it doesn't write camera parameters. I'd suggest to compute "fake" camera parameters in case ivmgr is NULL, what do you think?
I also noticed that this version in general opens all graspit world files with wrong camera parameters (zoomed in). I haven't looked at it yet but this should be a minor fix.

JenniferBuehler committed:
Added graspitConfig.cmake

JenniferBuehler committed:
EGPlanner working in headless mode from grasp_planning_graspit

JenniferBuehler committed:
Cleaned up graspitConfig.cmake and removed catkin flags

jvarley commented:
Can we make the graspitConfig.cmake a seperate PR.  I want to keep this concise.  

I am fine with the worldname for empty world. Good catch with the camera parameters.

JenniferBuehler committed:
Removed graspitConfig.cmake

JenniferBuehler commented:
Sure, I removed it.
I will add a suggestion about auto-computing camera parameters in headless mode, should I push it on this PR as well (it will get less concise because I'll put BB computation into it)?

JenniferBuehler committed:
Set IVmgr to world so camera parameters can be written. Added BB computation and camera parameters computation in for headless mode.

JenniferBuehler commented:
Ishh, now we updated something with the camera at the same time....

I've added BB computation and camera parameter computation in headless mode. 

The problem with parameters was the IVmgr was not set in World from GraspitCore. I fixed that too.

JenniferBuehler committed:
fixed merge conflict in world.cpp
